lib/dRep/dRepImpl.py

```diff
@@ -80,102 +77,321 @@ def run_dRep(self, ctx, params):
-    #        dprint('binnedContigs_wsObjInfo', binnedContigs_wsObjInfo)
 
 
         #
         ##
-        ### Run dRep dereplicate
+        ###
         ####
         #####
 
+        for bc in BinnedContigs.created_instances:
+            bc.calc_stats()
+
+
 
-        dRep_workDir = os.path.join(self.shared_folder, 'dRep_workDir_' + self.suffix)
+        #
+        ##
+        ### pool
+        ####
+        #####
+        
+ 
+        binsPooled_dir = os.path.join(_globals.run_dir, 'binsPooled')
+        os.mkdir(binsPooled_dir)
 
-        dRep_cmd = f'dRep dereplicate {dRep_workDir} -g {bins_dir}/*.fasta' 
-        if 'mode' in params and params['mode'] == 'local':
-            dRep_cmd += ' --debug --checkM_method taxonomy_wf'
-        dprint(f'Running CMD: {dRep_cmd}')
+        for bc in BinnedContigs.created_instances:
+            bc.pool_into(binsPooled_dir)
         
+        dprint("os.listdir(binsPooled_dir)", run={**locals(), **globals()})
+
+
+        #
+        ##
+        ### 
+        #### params
+        #####
+       
+
+        # flatten param groups (if any)
+
+        paramGrp_key_l = ['filtering', 'genome_comparison', 'clustering', 'scoring', 'warnings']
+
+        for paramGrp_key in paramGrp_key_l:
+            if paramGrp_key in params:
+                paramGrp_d = params[paramGrp_key]
+                for paramIndiv_key in paramGrp_d:
+                    params[paramIndiv_key] = paramGrp_d[paramIndiv_key]
+                params.pop(paramGrp_key)
+
+
+        #
+
+        dRep_params = ['--debug']
+
+
+        #
+        
+        if _globals.debug and 'processors' in params:
+            dRep_params.extend(['--processors', params.get('processors')])
+        else:
+            dRep_params.extend(['--processors', '8'])
+
+                
+        # extract non-default parameters
+
+        dRep_param_defaults = {
+                'length': 50000,
+                'completeness': 75,
+                'contamination': 25, 
+                'ignoreGenomeQuality': "False", 
+                'MASH_sketch': 1000,
+                'S_algorithm': 'ANImf',
+                'n_PRESET': 'normal',
+                'P_ani': 0.9,
+                'S_ani': 0.99,
+                'SkipMash': "False", 
+                'SkipSecondary': "False",
+                'cov_thresh': 0.1,
+                'coverage_method': 'larger',
+                'clusterAlg': 'average', 
+                'completeness_weight': 1,
+                'contamination_weight': 5,
+                'strain_heterogeneity_weight': 1, 
+                'N50_weight': 0.5,
+                'size_weight': 0,
+                'warn_dist': 0.25, 
+                'warn_sim': 0.98,
+                'warn_aln': 0.25,
+                'checkM_method': 'lineage_wf',
+                }
+
+        params_bool = [key for key in dRep_param_defaults if dRep_param_defaults[key] == 'False']
+
+        for key in dRep_param_defaults:
```

Comment:
I think this is more convoluted than it needs to be. If you want a dictionary of options from the user with a set of defaults, just merge the dictionaries:
```py
# Merge the user-supplied CLI parameters for dRep into our default parameters
drep_param_mapping = {**default_params, **user_params}
```